cmd/minikube/cmd/docker-env.go

```diff
@@ -366,6 +373,66 @@ docker-cli install instructions: https://minikube.sigs.k8s.io/docs/tutorials/doc
 			if err != nil {
 				exit.Error(reason.IfSSHClient, "Error with ssh-add", err)
 			}
+			if co.Config.KubernetesConfig.ContainerRuntime == constants.Containerd {
+				//create a  temporary directory
+				tmpDirName := "/tmp/minikube-nerdctld"
+				if err := os.Mkdir(tmpDirName, os.ModePerm); err != nil && !os.IsExist(err) {
+					exit.Error(reason.IfSSHClient, "Error with creating "+tmpDirName, err)
+				}
+
+				// for containerd runtime using ssh, we need to install nerdctld and nerdctl within os
+				if err := download.DownloadNerdctl(tmpDirName); err != nil {
+					exit.Error(reason.IfSSHClient, "Error with downloading nerdctl", err)
+				}
+				if err := download.DownloadNerdctld(tmpDirName); err != nil {
+					exit.Error(reason.IfSSHClient, "Error with downloading nerdctld", err)
+				}
+
+				//copy these binaries to the path of the containerd node
+				co := mustload.Running(ClusterFlagValue())
+				runner := co.CP.Runner
+				src := newRemotePath(filepath.Join(tmpDirName, "nerdctl"))
+				dst := newRemotePath("/usr/local/bin/nerdctl")
+				if err := runner.Copy(copyableFile(&co, src, dst)); err != nil {
+					exit.Error(reason.InternalCommandRunner, "Fail to copy file nerctl", err)
+				}
+
+				src = newRemotePath(filepath.Join(tmpDirName, "nerdctld"))
+				dst = newRemotePath("/usr/local/bin/nerdctld")
+				if err := runner.Copy(copyableFile(&co, src, dst)); err != nil {
+					exit.Error(reason.InternalCommandRunner, "Fail to copy file nerctld", err)
+				}
+
+				//and set 777 to these files
+				if out, err := runner.RunCmd(exec.Command("sudo", "chmod", "777", "/usr/local/bin/nerdctl", "/usr/local/bin/nerdctld")); err != nil {
+					exit.Error(reason.InternalCommandRunner, fmt.Sprintf("Fail to giver nerdctl proper permission: %s", out.Output()), err)
+				}
+
+				//remove /var/run/docker.sock
+				if out, err := runner.RunCmd(exec.Command("sudo", "rm", "-rf", "/var/run/docker.sock")); err != nil {
+					exit.Error(reason.InternalCommandRunner, fmt.Sprintf("Fail to delete docker.sock: %s", out.Output()), err)
+				}
+
+				//start nerdctld
+				nerdctldDaemonCommand := exec.Command("/bin/bash", "-c", "sudo env CONTAINERD_NAMESPACE=k8s.io nohup nerdctld --socket=/var/run/docker.sock >/dev/null 2>&1 &")
```

Comment:
There is proper packaging now - including systemd socket, so there is no need to start the daemon manually